### PR TITLE
Fire the LevelChange event after having enchanted an item

### PR DESCRIPTION
**Pull request by Cyborgmas**
The onEnchant method is responsible for reducing the xp level after an enchant.
However, it does not call addExperienceLevel which is the method that fires the LevelChange event.
This changes one line to make it pass through this method to fire the event. A test mod also logs the event, and it does fire after enchanting.
addExperienceLevel also plays a sound, but only if the level change is positive. SInce enchantment consumes levels, this should not happen unless mods play with the amount.

### CODE REVIEW

File: src/test/resources/META-INF/mods.toml [hunk 1]
```diff
@@ -65,7 +67,7 @@ loaderVersion="[28,)"
     modId="custom_plant_type_test"
 [[mods]]
     modId="chunk_data_event_save_null_world_test"
-[[dependencies.global_loot_test]]
+[[dependencies.level_change_event_on_enchant_test]]
     modId="forge"
```
Comment (LexManos) [inline src/test/resources/META-INF/mods.toml]: Why?

File: src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Mod("level_change_event_on_enchant_test")
+@Mod.EventBusSubscriber()
+public class LevelChangeEventOnEnchantTest {
+   private static final boolean ENABLE = true;
+   private static final Logger logger = LogManager.getLogger(LevelChangeEventOnEnchantTest.class);
+
+   @SubscribeEvent
+   public static void onPlayerLevelChange(PlayerXpEvent.LevelChange event)
+   {
+      if (!ENABLE) return;
```
Comment (pupnewfster) [inline src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java]: What is the point of this? If you want a flag like that for a one line body just use `if (ENABLED)`. Inverting it just to return seems silly.

File: src/test/resources/META-INF/mods.toml (hunk 1 above)
Comment (Cyborgmas) [reply src/test/resources/META-INF/mods.toml]: did this trying to get the test mod launching, but I had used the wrong run config

File: src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java (hunk 2 above)
Comment (Cyborgmas) [reply src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java]: copy pasted from another xp test mod.

File: src/test/resources/META-INF/mods.toml (hunk 1 above)
Comment (LexManos) [reply src/test/resources/META-INF/mods.toml]: Then don't include it in your PR. Only include intended changes.

File: src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java (hunk 2 above)
Comment (LexManos) [reply src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java]: Why not just add the event handler to that test mod? As they are both PlayerXPEvent...

Comment (Cyborgmas) [reply src/test/java/net/minecraftforge/debug/entity/player/LevelChangeEventOnEnchantTest.java]: ah yeah could have tested using the other mod instead huh... /oops

**Comment by Cyborgmas** [comment]
should i remove my test mod then? since it is unecessary?

**Comment by Cyborgmas** [comment]
Removed it, so only the patch needs to be merged. Did check with the old test mod it worked

**Pull request merged**